test/tests/functional/pbs_reservations.py

```diff
@@ -2452,3 +2452,74 @@ def test_shared_exclusive_job_not_in_same_rsv_vnode(self):
+        next_day_start = PBSLogUtils.convert_date_time(
+            next_day_start.strftime("%m/%d/%Y %H:%M:%S"))
+        next_day_end = next_day_start + 86400
+        self.scheduler.set_sched_config({'resv_confirm_ignore':
+                                         'dedicated_time'})
+
+        self.scheduler.add_dedicated_time(start=next_day_start,
+                                          end=next_day_end)
+        now = time.time()
+        rid = self.submit_reservation(user=TEST_USER, select='1:ncpus=1',
+                                      start=now + 5, end=now + 3600)
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, id=rid, offset=5)
+        resv_queue = rid.split('.')[0]
+        a = {'Resource_List.walltime': '00:02:00', ATTR_queue: resv_queue}
+        J1 = Job(attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.delete(jid1, wait=True)
+        a = {'Resource_List.walltime': '23:00:00', ATTR_queue: resv_queue}
+        J2 = Job(attrs=a)
+        J2.set_sleep_time(3600)
+        jid2 = self.server.submit(J2)
+        msg = "Job would cross dedicated time boundary"
+        attrs = {ATTR_comment: "Not Running: " + msg, ATTR_state: 'Q'}
+        self.server.delete(jid2, wait=True)
+        self.server.delete(rid, wait=True)
+
+        now = time.time()
+        rid = self.submit_reservation(user=TEST_USER, select='1:ncpus=1',
+                                      start=now + 5, end=now + 86400)
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, id=rid, offset=5)
+        resv_queue = rid.split('.')[0]
+        a = {'Resource_List.walltime': '00:02:00', ATTR_queue: resv_queue}
+        J1 = Job(attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.delete(jid1, wait=True)
+        a = {'Resource_List.walltime': '23:00:00', ATTR_queue: resv_queue}
+        J2 = Job(attrs=a)
+        J2.set_sleep_time(3600)
+        jid2 = self.server.submit(J2)
+        msg = "Job would cross dedicated time boundary"
+        attrs = {ATTR_comment: "Not Running: " + msg, ATTR_state: 'Q'}
+        self.server.expect(JOB, attrs, id=jid2)
+        self.server.delete(jid2, wait=True)
+        self.server.delete(rid, wait=True)
+
+        now = time.time()
+        self.scheduler.add_dedicated_time(start=now+10, end=now+300)
+        rid = self.submit_reservation(user=TEST_USER, select='1:ncpus=1',
+                                      start=now + 5, end=now + 60)
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, id=rid, offset=5)
+        resv_queue = rid.split('.')[0]
+        a = {'Resource_List.walltime': '00:02:00', ATTR_queue: resv_queue}
+        J1 = Job(attrs=a)
+        jid1 = self.server.submit(J1)
+        msg = "Dedicated time conflict"
```

Comment:
how is this scenario different from the previous one? Both create a reservation which overlaps with a dedicated time window. Both submit a job which will cross into the dedicated time window. So, why are the rejection reasons different? Am I misunderstanding something about this scenario?